hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/cocoapods/parse/PodlockParser.java

```diff
@@ -0,0 +1,210 @@
+                final Optional<DependencyId> dependencyId = parseDependencyId(podSource.getName());
+                if (dependencyId.isPresent()) {
+                    if (null != podSource.getGit() && podSource.getGit().contains("github")) {
+                        forgeOverrideMap.put(dependencyId.get(), Forge.COCOAPODS);
+                    } else if (null != podSource.getPath() && podSource.getPath().contains("node_modules")) {
+                        forgeOverrideMap.put(dependencyId.get(), Forge.NPM);
+                    }
+                }
+            }
+        }
+        return forgeOverrideMap;
+    }
+
+    private Forge getForge(final DependencyId dependencyId, final Map<DependencyId, Forge> forgeOverrides) {
+        if (forgeOverrides.containsKey(dependencyId)) {
+            return forgeOverrides.get(dependencyId);
+        }
+        return Forge.COCOAPODS;
+    }
+
+    private void processPod(final Pod pod, final Map<DependencyId, Forge> forgeOverrides, final LazyExternalIdDependencyGraphBuilder lazyBuilder) {
+        final String podText = pod.getName();
+        final Optional<DependencyId> dependencyIdMaybe = parseDependencyId(podText);
+        final String name = parseCorrectPodName(podText).orElse(null);
+        final String version = parseVersion(podText).orElse(null);
+        if (dependencyIdMaybe.isPresent()) {
+            final DependencyId dependencyId = dependencyIdMaybe.get();
+
+            final Forge forge = getForge(dependencyId, forgeOverrides);
+            final ExternalId externalId = externalIdFactory.createNameVersionExternalId(forge, name, version);
+
+            lazyBuilder.setDependencyExternalId(dependencyId, externalId);
+            lazyBuilder.setDependencyName(dependencyId, name);
+            lazyBuilder.setDependencyVersion(dependencyId, version);
+
+            for (final String child : pod.getDependencies()) {
+                final Optional<DependencyId> childId = parseDependencyId(child);
+                if (childId.isPresent()) {
+                    if (!dependencyId.equals(childId.get())) {
+                        lazyBuilder.addParentWithChild(dependencyId, childId.get());
+                    }
+                }
+            }
+
+            final Optional<String> superName = parseSuperPodName(podText);
+            if (superName.isPresent()) {
+                final DependencyId superDependencyId = new NameDependencyId(superName.get());
+                final Forge superForge = getForge(superDependencyId, forgeOverrides);
+                final ExternalId superExternalId = externalIdFactory.createNameVersionExternalId(superForge, superName.get(), version);
+
+                lazyBuilder.setDependencyExternalId(superDependencyId, superExternalId);
+                lazyBuilder.setDependencyName(superDependencyId, name);
+                lazyBuilder.setDependencyVersion(superDependencyId, version);
+
+                if (!dependencyId.equals(superDependencyId)) {
```

Comment:
If the super is present doesn't that mean the dependency is always going to be the super? Do you have an example where that is not true?